### ISSUE TITLE
Prometheus exporter: allow to work via the application's port

### ISSUE THREAD

**Issue by SergeyKanzhelev** [open-telemetry/opentelemetry-dotnet#65]
Original: https://github.com/census-instrumentation/opencensus-csharp/issues/65

In some environments opening separate port for Prometheus exporter will not work. Especially for sandboxed environments like Azure Web Apps. Re-using app's port and allow Prometheus to work as a middleware will help to enable those environments.

**Comment by MikeGoldsmith** [comment]
@SergeyKanzhelev I'm working on this, but don't have access to self-assign.

**Comment by MikeGoldsmith** [comment]
I've pushed first PR to break the Exporters dependency on HttpListener. I'll work on additional PRs to add support for other methods of exposing the `/metrics` endpoint, eg ASP.NET Core.

**Comment by tomstreet** [comment]
Hi just ran into this. It would be really good to have the logic that writes the prometheus metrics response to a stream as something separate that can be injected anywhere - this would make it nice and reusable. That way it doesn't have to be a middleware, but can be easily injected into a middleware to get that kind of functionality.

**Comment by MikeGoldsmith** [comment]
Hi @tomstreet - the work to do that is in the open PR #541. Hopefully it won't be before we can merge into master.

**Comment by cijothomas** [comment]
This can be closed as #541 is merged
